src/services/fluence-aurora-connector/modules/fluence_aurora_connector/src/main.rs

```diff
@@ -204,249 +74,61 @@ pub fn poll_deals(api_endpoint: String, address: String, from_block: String) ->
-        return DealChangedResult::error(deal_id, err.to_string());
-    }
-
-    let address = format!("0x{}", deal_id);
-    let to_block = get_to_block(&from_block);
-    let result = poll(
-        api_endpoint,
-        address,
-        from_block,
-        to_block.clone(),
-        DealChangedData::topic(),
-    );
-    match result {
-        Err(err) => return DealChangedResult::error(deal_id, err.to_string()),
-        Ok(deals) => {
-            let changed_deals = parse_deals::<DealChangedData, DealChanged>(deals);
-            DealChangedResult::ok(deal_id, changed_deals, to_block)
-        }
-    }
-}
-
-#[derive(Debug)]
-#[marine]
-pub struct DealUpdate {
-    deal_info: DealInfo,
-    from_block: String,
-}
-
-#[derive(Debug)]
-#[marine]
-pub struct DealInfo {
-    worker_id: String,
-    deal_id: String,
-}
-
-#[marine]
-pub struct DealUpdatedBatchResult {
-    success: bool,
-    /// optional error
-    error: Vec<String>,
-    /// optional result (present if success is true)
-    result: Vec<DealChanged>,
-    /// The request checked blocks from `from_block` to `to_block`
-    to_block: String,
-    /// Return deal info to be able to find which deal to update
-    deal_info: DealInfo,
-}
-
-impl DealUpdatedBatchResult {
-    fn ok(to_block: String, deal_info: DealInfo, update: DealChanged) -> Self {
-        Self {
-            success: true,
-            error: vec![],
-            result: vec![update],
-            to_block,
-            deal_info,
-        }
-    }
-
-    fn error(to_block: String, deal_info: DealInfo, err: String) -> Self {
-        Self {
-            success: false,
-            error: vec![err],
-            result: vec![],
-            to_block,
-            deal_info,
-        }
-    }
-}
+/// REMOVAL
+/// I have removed `poll_deal_changed` because no one is using it
+/// But it makes overall code more complex
+// #[marine]
+// pub fn poll_deal_changed(
+//     api_endpoint: String,
+//     deal_id: String,
+//     from_block: String,
+// ) -> DealChangedResult
 
 #[marine]
-pub struct DealsUpdatedBatchResult {
-    result: Vec<DealUpdatedBatchResult>,
-    success: bool,
-    error: Vec<String>,
-}
-
-impl DealsUpdatedBatchResult {
-    fn ok(result: Vec<DealUpdatedBatchResult>) -> Self {
-        Self {
-            success: true,
-            error: vec![],
-            result,
-        }
-    }
-
-    fn error(err: String) -> Self {
-        Self {
-            success: false,
-            error: vec![err],
-            result: vec![],
-        }
-    }
-}
-
-#[marine]
-pub fn poll_deals_latest_update_batch(
-    api_endpoint: String,
-    deals: Vec<DealUpdate>,
-) -> DealsUpdatedBatchResult {
+/// RENAMED
+/// old name `poll_deals_latest_update_batch` was confusing, and using `update` wording which I removed
+/// new name `poll_deal_changes` is shorter and is similar to `poll_deal_created`
+pub fn poll_deal_changes(api_endpoint: String, deals: Vec<DealUpdate>) -> MultipleDealsChanged {
```

Comment:
Shouldn't it be `poll_dealS_changes` then?

Reply:
Nah, usually one says something like "this is a pan of pig tails", not "this is a pan of pigs tails", even though multiple pigs certainly were involved.

Same here, all we care about is a bunch of deal changes.